pkg/util/cert_pool.go

```diff
@@ -21,19 +21,22 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
+	"time"
 )
 
 // CertPool is a set of certificates.
 type certPool struct {
 	certificatesHashes map[[32]byte]struct{}
 	certificates       []*x509.Certificate
+	filterExpiredCerts bool
 }
 
 // NewCertPool returns a new, empty CertPool.
-func newCertPool() *certPool {
+func NewCertPool(filterExpiredCerts bool) *certPool {
 	return &certPool{
 		certificatesHashes: make(map[[32]byte]struct{}),
 		certificates:       make([]*x509.Certificate, 0),
+		filterExpiredCerts: filterExpiredCerts,
 	}
 }
```

Comment:
suggestion: I appreciate you making a change so that we don't break `ValidateAndSanitizePEMBundle`!

I don't think we need to make this specific change - `ValidateAndSanitizePEMBundle` is in the `util` package and so it can call `newCertPool` without an issue. It'd be my preference to keep certPool entirely private so it's easier for us to change later!

I should've suggested this in my first review, but how about keeping `newCertPool` private and adding something like this:

```go
type ValidateAndSanitizeOptions struct {
    FilterExpired bool
}

func ValidateAndSanitizePEMBundleWithOptions(data []byte, opts ValidateAndSanitizeOptions) ([]byte, error) {
    // implementation here ...
}

func ValidateAndSanitizePEMBundle(data []byte) ([]byte, error) {
    opts := ValidateAndSanitizeOptions{
        FilterExpired: false,
    }

    return ValidateAndSanitizePEMBundleWithOptions(data, opts)
}
```

What do you think? 😁

Reply:
Okay, I see. I have made the update.